### PR TITLE
tests: add test to verify unmatched signatures does not write to dataset

### PR DESCRIPTION
**Pull request by norg**
Suricata Verify test related to https://redmine.openinfosecfoundation.org/issues/5576

### CODE REVIEW

File: tests/datasets-validate-postmatch/test.rules [hunk 1]
```diff
@@ -0,0 +1 @@
+alert http any any -> $HOME_NET any (msg:"HTTP learning"; flow:established,to_client; http.content_type; content:"noone"; http.server; content:"ECS"; fast_pattern; dataset:set,http,type string,state output/http.intel; sid:2; rev:1; priority:2;)
```
Comment (inashivb) [inline tests/datasets-validate-postmatch/test.rules:1]: We should keep the filename to `http.intel` only as `output` dir doesn't exist initially and is written everytime we run the test.

**Review by inashivb — CHANGES_REQUESTED** [review]
Not moved forward with this yet. Was just trying and found a complaint of rule failure by Suricata. :)

**Review by inashivb — APPROVED** [review]
The earlier review was incorrect. This is fine but needs tweaks if run with gdb

**Comment by catenacyber** [comment]
cf https://github.com/OISF/suricata/pull/8114

**Comment by victorjulien** [comment]
PR is too old to restart the CI, can you resubmit?

**Review by catenacyber — CHANGES_REQUESTED** [review]
Rebase + Resubmit was requested

**Pull request closed without merging**